pkg/api/rest/delete.go

```diff
@@ -28,7 +30,11 @@ import (
 // API conventions.
 type RESTDeleteStrategy interface {
 	runtime.ObjectTyper
+}
 
+// GracefulDeleteStrategy must be implemented by the registry that supports
+// graceful deletion.
+type GracefulDeleteStrategy interface {
```

Comment:
"GracefulDeleter" would be the go-ish name, but I guess this has more symmetry with the other strategy interfaces. RESTGracefulDeleteStrategy would be even more symmetric. Up to you.